pkg/daemon/ceph/osd/kms/kms.go

```diff
@@ -169,19 +193,18 @@ func ValidateConnectionDetails(clusterdContext *clusterd.Context, securitySpec c
 		if err != nil {
 			return errors.Wrap(err, "failed to validate vault connection details")
 		}
-	}
-
-	// Validate potential token Secret presence
-	if securitySpec.KeyManagementService.IsTokenAuthEnabled() {
-		kmsToken, err := clusterdContext.Clientset.CoreV1().Secrets(ns).Get(ctx, securitySpec.KeyManagementService.TokenSecretName, metav1.GetOptions{})
-		if err != nil {
-			return errors.Wrapf(err, "failed to fetch kms token secret %q", securitySpec.KeyManagementService.TokenSecretName)
-		}
 
-		// Check for empty token
-		token, ok := kmsToken.Data[KMSTokenSecretNameKey]
-		if !ok || len(token) == 0 {
-			return errors.Errorf("failed to read k8s kms secret %q key %q (not found or empty)", KMSTokenSecretNameKey, securitySpec.KeyManagementService.TokenSecretName)
+		secretEngine := securitySpec.KeyManagementService.ConnectionDetails[VaultSecretEngineKey]
+		switch secretEngine {
+		case VaultKVSecretEngineKey:
```

Comment:
An error case if this is not the case?

Reply:
No we also support the transit engine, so we don't need to error here.